lib/ncr/work_order_reapproval_checker.rb

```diff
@@ -6,13 +6,21 @@ def initialize(work_order)
       @work_order = work_order
     end
 
-    def current_user
-      work_order.modifier
+    def requires_budget_reapproval?
+      work_order.approved? &&
+        work_order.requires_approval? &&
+        !budget_approver? &&
+        (amount_increased? || protected_fields_changed?)
     end
 
-    def previous_val(key)
-      changes = work_order.previous_changes || {}
-      changes[key.to_s].try(:first)
+    private
+
+    def budget_approver?
+      work_order.budget_approvers.include?(current_user)
+    end
```

Comment:
I'm not positive that the behavior has actually been changed here. I think this method should roughly read:

```ruby
def budget_approver?
  Ncr.budget_mailboxes_and_their_delegates.include?(current_user)
end
```